### PR TITLE
[NU-1772] Deployment comment validation

### PR DESCRIPTION
**Pull request by mgoworko**
## Describe your changes

When editing or deleting comment assigned to ScenarioDeployed activity Nu should follow the same validation rules like when creating the comment. 

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Comment by mgoworko** [comment]
I added commit with refactor of non-empty comment handling - the autogenerated apply method was still used in a few places, detected by Cypress tests.

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/config/FeatureTogglesConfigTest.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ class FeatureTogglesConfigTest extends AnyFunSuite with Matchers {
 
   test("should be ok with no deploymentCommentSettings") {
     val config = ConfigFactory
-      .parseResources("config/business-cases/simple-streaming-use-case-designer.conf")
+      .parseResources("config/business-cases/category-used-more-than-once-designer.conf")
```
Comment (mk-software-pl) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/config/FeatureTogglesConfigTest.scala:15]: why?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala [hunk 2]
```diff
@@ -535,26 +542,84 @@ class ScenarioActivityApiHttpService(
         scenarioActivityRepository.editComment(
           scenarioId,
           ScenarioActivityId(request.scenarioActivityId),
-          request.commentContent
+          validateComment(_, request.commentContent),
```
Comment (mk-software-pl) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala:545]: I'm not convinced with the current solution.

I see the following problems:
1. The validation should be done in the core layer and the repository layer should not be bothered with this kind of validation. After we validate some comment we should (at the core layer) produce an instance that holds the invariant that proves (to the repository layer) the instance is validated. 

2. The `addNewComment` method adds sth like a simple/regular comment. The `editComment` method modifies a simple/regular comment and deployment comment (this one has different validation rules). We don't have such distinction at the type level, but IMO we should have it. 

3. Comment smart constructor should have a validation logic encapsulated. IMO too much service/core logic is encapsulated in the repository.

**Review by mk-software-pl — CHANGES_REQUESTED** [review]

**Review by mk-software-pl — APPROVED** [review]
let's address all the comments in a separate PR

**Pull request merged**